executor/builder.go

```diff
@@ -1185,10 +1184,14 @@ func (b *executorBuilder) getStartTS() uint64 {
 	}
 
 	startTS := b.ctx.GetSessionVars().SnapshotTS
-	if startTS == 0 && b.ctx.Txn() != nil {
+	if startTS == 0 && b.ctx.Txn().Valid() {
 		startTS = b.ctx.Txn().StartTS()
 	}
 	b.startTS = startTS
+	if b.startTS == 0 {
+		// The the code should never run here if there is no bug.
+		log.Error(errors.ErrorStack(ErrGetStartTS))
```

Comment:
The error of `ErrGetStartTS` originally happened here. So I think we needn't `ErrorStack`.

Reply:
>  // The the code should never run here if there is no bug.

@zimulala

Reply:
This log should never happen... I add it just in case there maybe bugs of the code

Reply:
And print the stack info for us to find the bug

Reply:
Done